core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParserErrorHandling.java

```diff
@@ -160,7 +160,9 @@ private static Stream<Arguments> statements()
                 Arguments.of("SELECT * FROM t FOR TIMESTAMP AS OF TIMESTAMP WHERE",
                         "line 1:52: mismatched input '<EOF>'. Expecting: <expression>"),
                 Arguments.of("SELECT * FROM t FOR VERSION AS OF TIMESTAMP WHERE",
-                        "line 1:50: mismatched input '<EOF>'. Expecting: <expression>"));
+                        "line 1:50: mismatched input '<EOF>'. Expecting: <expression>"),
+                Arguments.of("SELECT ROW(DATE '2022-10-10', DOUBLE 12.0)",
```

Comment:
This is reporting the following position:

```
SELECT ROW(DATE '2022-10-10', DOUBLE 12.0)
                                     ^
```

How are all of those valid tokens?

Reply:
See the explanation above. It's a similar problem.